### PR TITLE
Videos playlist

### PR DESCRIPTION
**Pull request by Naveenchandar**
* Playlist route added
* Create, edit, update and delete playlist
* playlist modal added
* list of playlist added
* playlist videos added

playlist_modal

create_playlist

update_playlist

playlists

### CODE REVIEW

**Review by aniketsingh13 — APPROVED** [review]
ui look great.
code is also easy to understand

File: src/components/sidebar/index.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-    { path: '/watchlater', name: 'Watch later' },
-    { path: '/', name: 'Liked videos' },
-    { path: '/', name: 'History' }
+    { path: '/', name: 'Home', icon: <AiOutlineHome /> },
+    { path: '/explore', name: 'Explore', icon: <MdOutlineExplore /> },
+    { path: '/playlists', name: 'Playlists', icon: <RiPlayListLine /> },
+    { path: '/watchlater', name: 'Watch later', icon: <MdOutlineWatchLater /> },
+    { path: '/', name: 'Liked videos', icon: <BiLike /> },
+    { path: '/', name: 'History', icon: <MdOutlineHistory /> }
 ];
```
Comment (rohanmathur91) [inline src/components/sidebar/index.js]: you can move this into separate file say data or staticData

File: src/components/videos/index.js [hunk 2]
```diff
@@ -27,7 +32,7 @@ export function Vidoes() {
             }
             dispatch({ type: 'LOADING_VIDEOS', payload: false });
         })()
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (rohanmathur91) [inline src/components/videos/index.js:35]: suggestion: try to find out any other solution to this problem as it's not a good practice to add this comment

File: src/reducer/videos.js [hunk 3]
```diff
@@ -3,7 +3,9 @@ const initialState = {
     filterVideoList: [],
     error: '',
     loading: false,
-    activeChip: 'all'
+    activeChip: 'all',
+    videoPlaylistInfo: {},
+    // playlistVideos: []
```
Comment (rohanmathur91) [inline src/reducer/videos.js]: remove commented code

File: src/reducer/videos.js [hunk 4]
```diff
@@ -21,6 +23,12 @@ function videosReducer(state = initialState, action) {
                 return { ...state, filterVideoList: filterResult };
             }
             return { ...state, filterVideoList: state.videoList };
+        case 'VIDEO_ACTIVE_PLAYLIST_MODAL': {
+            return { ...state, videoPlaylistInfo: action.payload };
+        }
+        // case 'VIDEOS_PLAYLIST': {
+        //     return {...state, playlistVideos: [...state.playlistVideos, { name, videos: data }]}
+        // }
```
Comment (rohanmathur91) [inline src/reducer/videos.js]: remove commented code

File: src/reducer/playlist.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                if (id === playlistId) {
+                    return { ...item, videos: videos.filter(({ _id }) => _id !== videoId) };
+                }
+                return { ...item };
+            });
+            return { ...state, playlists: result };
+        }
+        default: {
+            return initialState;
+        }
```
Comment (rohanmathur91) [inline src/reducer/playlist.js:92]: throw an error saying `action type not found` instead of returning initial state

**Review by rohanmathur91 — APPROVED** [review]
Liked that you have provided default values wherever you have destructed.
Added a few comments, the rest looks good.

**Pull request merged**